src/app/mesh/qgsmaptooleditmeshframe.cpp

```diff
@@ -902,6 +967,10 @@ void QgsMapToolEditMeshFrame::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
         {
           addVertex( mFirstClickPoint, e->mapPointMatch() );
           mCadDockWidget->setPoints( QList<QgsPointXY>() << mFirstClickPoint << mFirstClickPoint );
+
+          // after addition select the vertex for easier editing of Z value
+          mCurrentVertexIndex = closeVertex( mapPoint );
+          select( mapPoint, e->modifiers(), tolerance );
```

Comment:
I'm not sure the modifiers should be forwarded here. It might not make a difference now, but if for example `ctrl+double click` gets supported, it will cause the new vertex to not be selected.

Reply:
This was originally there so I did not want to change it ...